h2integrate/finances/finances.py

```diff
@@ -44,32 +45,43 @@ def setup(self):
         self.target_dollar_year = plant_config["finance_parameters"]["cost_adjustment_parameters"][
             "target_dollar_year"
         ]
+        plant_life = self.plant_life = int(self.options["plant_config"]["plant"]["plant_life"])
 
         for tech in tech_configs:
             self.add_input(f"capex_{tech}", val=0.0, units="USD")
             self.add_input(f"opex_{tech}", val=0.0, units="USD/year")
+            self.add_input(f"varopex_{tech}", val=0.0, shape=plant_life, units="USD/year")
+
             self.add_discrete_input(f"cost_year_{tech}", val=0, desc="Dollar year for costs")
 
             self.add_output(f"capex_adjusted_{tech}", val=0.0, units="USD")
             self.add_output(f"opex_adjusted_{tech}", val=0.0, units="USD/year")
+            self.add_output(f"varopex_adjusted_{tech}", val=0.0, shape=plant_life, units="USD/year")
 
         self.add_output("total_capex_adjusted", val=0.0, units="USD")
         self.add_output("total_opex_adjusted", val=0.0, units="USD/year")
+        self.add_output("total_varopex_adjusted", val=0.0, shape=plant_life, units="USD/year")
 
     def compute(self, inputs, outputs, discrete_inputs, discrete_outputs):
         total_capex_adjusted = 0.0
         total_opex_adjusted = 0.0
+        total_varopex_adjusted = np.zeros(self.plant_life)
         for tech in self.options["tech_configs"]:
             capex = float(inputs[f"capex_{tech}"][0])
             opex = float(inputs[f"opex_{tech}"][0])
+            varopex = inputs[f"varopex_{tech}"][0]
```

Comment:
This `[0]` means that we're only grabbing the first entry in this array that's length `plant_life`. Is this intentional? This means that all entries will have the same value in the array and won't vary year-to-year.

Reply:
In general, I'd suggest not using the `[0]` indexing when dealing OM variables and only doing so when we absolutely need a `float` value and not an `array`. All variables (inputs and outputs) are represented as numpy arrays even if it's just an array of length 1. This is slick because all the normal numpy operations work regardless of if it's length 1, or size (360, 120, 4) (for example).